src/sincere_singularities/data/db.py

```
@@ -0,0 +1,225 @@
+        # Check if the restaurant already exists
+        if not self.db[collection].find_one({"name": name}):
+            current_time = datetime.now(utc_timezone)
+            data["created_at"] = current_time
+            data["updated_at"] = current_time
+            self.db[collection].insert_one(data)
+
+    def add_many(self, collection: str, datas: Iterable[Any]) -> None:
+        """Add many elements
+
+        Args:
+            collection (str): Collection name
+            datas (Iterable[Any]): list of Datas to add
+
+        Returns:
+            None
+        """
+        for data in datas:
+            self.add_element(collection, data)
+
+    def delete_all(self, collection: str) -> None:
+        """Delete all elements
+
+        Args:
+            collection (str): Collection name
+
+        Returns:
+            None
+        """
+        if not self.connected:
+            raise ConnectError("Not connected to the database")
+        self.db[collection].delete_many({})
+
+    def delete_many(self, collection: str, datas: Iterable[Any]) -> None:
+        """Delete many elements
+
+        Args:
+            collection (str): Collection name
+            datas (Iterable[Any]): list of datas to delete
+
+        Returns:
+            None
+        """
+        if not self.connected:
+            raise ConnectError("Not connected to the database")
+
+        for data in datas:
+            if self.show_one(collection, data):
+                self.db[collection].delete_one(data)
+
+    def delete_one(self, collection: str, data: dict[str, Any]) -> None:
+        """Delete one element
+
+        Args:
+            collection (str): Collection name
+            data (dict[str, Any]): Data to delete
+
+        Returns:
+            None
+        """
+        if not self.connected:
+            raise ConnectError("Not connected to the database")
+        if self.show_one(collection, data):
+            self.db[collection].delete_one(data)
```

Maybe raise an exception if it didn't exist in the first place? (also in `delete_many`)